KS-18-02-playing-dice

DenisGubernatorov opened the pull request:
1) Базовый класс PlayingDiceL18 с базовым количеством граней и пустой реализацией метода отвечающего за бросок 2) Классы наследники с 4,6 и 8 гранями. В зависимости от класса метод getThrowResult берет рандомное число в своем диапазоне

DenisGubernatorov committed:
KS-18-02-playing-dice
1) Базовый класс PlayingDiceL18 с базовым количеством граней и пустой реализацией метода отвечающего за бросок
2) Классы наследники с 4,6 и 8 гранями. В зависимости от класса метод getThrowResult берет рандомное число в своем диапазоне

ievetrovdev commented on src/main/kotlin/lesson18/lesson18_task2.kt:
```diff
@@ -1 +1,36 @@
 package lesson18
+
+open class PlayingDiceL18 {
```
В создании экземпляра базового класса нет потребности, а это значит, что лучше сделать его абстрактным.

ievetrovdev commented on src/main/kotlin/lesson18/lesson18_task2.kt:
```diff
@@ -1 +1,36 @@
+class FourEdgeDice : PlayingDiceL18() {
+    override val maxEdgeCount = 4
+
+    override fun getThrowResult() {
```
Мы можем не переопределять этот метод в каждом классе, а использовать его реализацию по умолчанию, и оставить только переопределение количества граней

DenisGubernatorov committed:
KS-18-02-playing-dice
1) Класс PlayingDiceL18 сделан абстрактным с дефолтной реализацией метода getThrowResult.
2) Классы наследники переопределяют только количество граней кубиков

ievetrovdev commented on src/main/kotlin/lesson18/lesson18_task2.kt:
```diff
@@ -1 +1,26 @@
+}
+
+fun main() {
+    val dices = mutableListOf(FourEdgeDice(), SixEdgeDice(), EightEdgeDice())
```
Давай еще принудительно укажем тип списка для демонстрации работы полиморфизма

DenisGubernatorov committed:
KS-18-02-playing-dice
1) Принудительно указан тип объектов для списка dices
